### PR TITLE
Add scorecard parser integration

### PR DESCRIPTION
**Pull request by lumjjb**
This PR implements scorecard with the new interface:
- Define fields of placeholder (another PR to rename Placeholder to another name right after for ease of review)
- Add parsing to predicates for scorecards
- Changing of DocumentParser interface to take `*Placeholder` instead of `[]assembler.Placeholder`
- Unfreeze test for scorecards parser

### CODE REVIEW

**Comment by lumjjb** [comment]
need to change this due to conflicting changes from rebasing https://github.com/guacsec/guac/pull/499, hold on review

File: pkg/ingestor/parser/scorecard/parser_scorecard.go [hunk 1]
```diff
@@ -55,95 +54,68 @@ func (p *scorecardParser) Parse(ctx context.Context, doc *processor.Document) er
...
-	return mnNode
-}
-
-func getArtifactNode(s *sc.JSONScorecardResultV2) assembler.ArtifactNode {
-	return assembler.ArtifactNode{
-		Name:   sourceUri(s.Repo.Name),
-		Digest: hashToDigest(s.Repo.Commit),
+	srcInput := model.SourceInputSpec{
+		// assuming scorecards is only git
+		Type:      "git",
```
Comment (mihaimaruseac) [inline pkg/ingestor/parser/scorecard/parser_scorecard.go:99]: Can we get it from `s.Repo.Name`?

**Review by mihaimaruseac — APPROVED** [review]
LGTM in principle, though I think we might need to change interfaces. But if this lands before my PR for the GQL client, I'll change the types accordingly.

**Comment by lumjjb** [comment]
Ref: https://github.com/ossf/scorecard/issues/2711

File: pkg/ingestor/parser/scorecard/parser_scorecard.go (hunk 1 above)
Comment (lumjjb) [reply pkg/ingestor/parser/scorecard/parser_scorecard.go:103]: hmm you mean like parsing and saying "github.com" and "gitlab.com" is git and "www.mercurial-scm.org" is "hg"?

Comment (mihaimaruseac) [reply pkg/ingestor/parser/scorecard/parser_scorecard.go:103]: OH, I thought `s.Repo` contains a full URL like `git@github.com:guacsec/guac`, but if it is not then nothing to do

File: pkg/ingestor/parser/scorecard/parser_scorecard.go [hunk 2]
```diff
@@ -20,25 +20,25 @@ import (
 	"encoding/json"
 	"fmt"
 	"strings"
+	"time"
 
 	"github.com/guacsec/guac/pkg/assembler"
+	"github.com/guacsec/guac/pkg/assembler/graphql/model"
```
Comment (mihaimaruseac) [inline pkg/ingestor/parser/scorecard/parser_scorecard.go:26]: Will need to change import to be the same as in #522, but the structs would stay similar (just in different package)

**Review by pxp928 — APPROVED** [review]
LGTM

**Pull request merged**